utilities/dataflow_hcapi_to_gcs_pubsub/dataflow_hcapi_to_gcs_pubsub.py

```diff
@@ -0,0 +1,380 @@
+            errorMessage = "process to transform pub/sub record from HCAPI failed due to : {}".format(str(type(error).__name__)+" --> "+ str(error))
+            rawMessage = dict()
+            rawMessage['errorMessage'] = errorMessage
+            rawMessage['rawMessage']= str(element)
+            yield pvalue.TaggedOutput(self.failTag,rawMessage)
+    
+class writeToGCS(beam.DoFn):
+    """ Class to write raw and errored messages to GCS """
+    def process(self, element, bucket, messageType, topic=None):
+        """
+         write raw / error messages to respective bucket
+        :param str element: PCollection of message from convert kafka record function
+        :param str bucket: bucket name
+        :param str messageType: message type raw or error message
+        :return: None message written to cloud storage
+        :rtype: None
+        """
+        #TO DO below logging only for testing purpose delete when deploying to prod
+        logging.info("printing raw element : {}".format(str(json.dumps(element))))
+        
+        if messageType :
+            folderName= str(messageType)+"_messages"
+        else:
+            folderName= "unidentified"
+        
+        #converting incoming element to JSON
+        newElement = json.loads(json.dumps(element,separators=(',', ':')))
+        if topic:
+            topicName = str(topic)
+        elif 'topic' in newElement.keys():
+            topicName = str(newElement['topic'])
+        else:
+            logging.info("topic name not mentioned or not present in data while writing to GCS")
+        
+        if topicName:
+            if folderName and bucket:
+                strJsonElement = str(newElement).encode('utf-8')
+                #Building GCS path based on current timestamp
+                currentTimestamp = datetime.now().strftime("%Y-%m-%d-%H%M%S")
+                gcsPath = f"gs://{bucket}/{folderName}/{topicName}/{str(currentTimestamp[:10])}/{str(currentTimestamp.replace('-',''))}.json"
+                file = io.gcsio.GcsIO().open(filename=gcsPath, mode='w', mime_type='application/json')
+                write_file = file.write(strJsonElement)
+                logging.info("successfully written message to cloud storage at :- {}".format(gcsPath))                
+            else:
+                logging.error("foldername and bucket not mentioned")
+        else:
+            logging.error("topic name not mentioned or not present in topic")
+    
+            
+class buildPubSubMesage(beam.DoFn):
+    """ Beam DoFn to transform raw data and create careplan FHIR resource"""
+    #beam output tag to mark transformed element as pass and fail
+    def process(self, element):
+        """ Function to build PubSub message as per:
+        https://beam.apache.org/releases/pydoc/2.29.0/apache_beam.io.gcp.pubsub.html#module-apache_beam.io.gcp.pubsub """
```

Comment:
```suggestion
        https://beam.apache.org/releases/pydoc/2.53.0/apache_beam.io.gcp.pubsub.html#module-apache_beam.io.gcp.pubsub """
```

Let's use a more recent version of the PyDocs.